R/graded.R

```diff
@@ -650,15 +650,15 @@ fail_if_code_feedback <- function(
     return(grade_code_is_empty())
   }
 
-  if (is_placeholder(solution_code, ".solution_code")) {
-    solution_code <- get_from_env(".solution_code", env)
-    assert_object_found_in_env(solution_code, env, "fail_if_code_feedback", throw_grade = FALSE)
+  if (is_placeholder(solution_code_all, ".solution_code_all")) {
+    solution_code_all <- get_from_env(".solution_code_all", env)
+    assert_object_found_in_env(solution_code_all, env, "fail_if_code_feedback", throw_grade = FALSE)
     if (
-      is_placeholder(solution_code) ||
-        is_missing(solution_code) ||
-        is.null(solution_code) ||
-        length(solution_code) == 0 ||
-        !nzchar(solution_code)
+      is_placeholder(solution_code_all) ||
+        is_missing(solution_code_all) ||
+        is.null(solution_code_all) ||
+        length(solution_code_all) == 0 ||
+        !nzchar(solution_code_all)
```

Comment:
```suggestion
        any(!nzchar(solution_code_all))
```

Surprisingly `nzchar()` works on a list

```r
!nzchar(list('a', 'b', ''))
#> [1] FALSE FALSE  TRUE
```

but we need to make sure at least one solution has some code in it. (That seems unlikely but it is certainly the safest thing for this test.)